File: FlowCrypt/src/main/java/com/flowcrypt/email/security/model/PrivateKeySourceType.java
/*
 * Business Source License 1.0 © 2017 FlowCrypt Limited (tom@cryptup.org).
 * Use limitations apply. See https://github.com/FlowCrypt/flowcrypt-android/blob/master/LICENSE
 * Contributors: DenBond7
 */

package com.flowcrypt.email.security.model;

/**
 * @author DenBond7
 *         Date: 13.05.2017
 *         Time: 15:09
 *         E-mail: DenBond7@gmail.com
 */

public enum PrivateKeySourceType {
    BACKUP("backup"),
    NEW("new"),
    IMPORT("import");

    private final String text;

    /**
     * @param text
     */
    private PrivateKeySourceType(final String text) {
        this.text = text;
    }

    /* (non-Javadoc)
     * @see java.lang.Enum#toString()
     */
    @Override
    public String toString() {
        return text;
    }
}

File: FlowCrypt/src/main/java/com/flowcrypt/email/security/model/PrivateKeySourceType.kt
/*
 * © 2016-present FlowCrypt a.s. Limitations apply. Contact human@flowcrypt.com
 * Contributors: DenBond7
 */

package com.flowcrypt.email.security.model

/**
 * @author Denys Bondarenko
 */
enum class PrivateKeySourceType constructor(private val text: String) {
  BACKUP("backup"),
  NEW("new"),
  IMPORT("import");

  override fun toString(): String {
    return text
  }
}
